### ISSUE TITLE
Ending call no longer redirects to application context

### ISSUE THREAD

**Issue by lochner** [igniterealtime/openfire-pade-plugin#372]
Since version 1.4.2-SNAPSHOT, built on 2021-10-14, clicking the red phone to end a call redirects to / instead of the welcome page at the configured (or /ofmeet/ default) application context.

A work-around for version 1.5.1 is to copy the "BAO" fix from four lines higher into line 23091 of the deployed <Openfire>\plugins\pade\classes\jitsi-meet\libs\app.bundle.min.js:

-a.pathname = "/", window.location.assign(a.toString())
+a.pathname = "/" + (location.href.split("/").length == 5 ? location.href.split("/")[3] : ""), window.location.assign(a.toString())

**Comment by deleolajide** [comment]
It should be fixed with this commit - https://github.com/igniterealtime/openfire-pade-plugin/commit/6f36d5ace95a3b4fee8b7925e4ad57d05da5faee

Please check latest nightly build and re-open if issue has not been fixed. Thanks.

**Comment by lochner** [comment]
Now also tested the latest 1.5.2-SNAPSHOT, built on 2021-12-12, and confirmed that the problem still exists.&nbsp; It's also still fixed by the change above.

Thank you for looking into this!

**Comment by deleolajide** [comment]
I am unable to re-produce this issue. Can you confirm it happens to you at https://pade.chat:5443/ofmeet

**Comment by lochner** [comment]
I cannot reproduce the problem on your test site.

However, the code is different.&nbsp; In https://pade.chat:5443/ofmeet/libs/app.bundle.min.js, from line 25900 onwards:
```
}, Yf.STICKY)), n()["features/base/config"].enableWelcomePage && setTimeout((() => {
    t(nB("/" + (location.href.split("/").length == 5 ? location.href.split("/")[3] : "")))	// BAO
}), e.showThankYou ? 3e3 : 500)
```
In constrast to app.bundle.min.js from 1.5.2-SNAPSHOT, from line 23086 onwards:
```
})), n()["features/base/config"].enableWelcomePage && setTimeout((() => {
    t(("/" + (location.href.split("/").length == 5 ? location.href.split("/")[3] : ""), (e, t) => {	// BAO
        const {
            locationURL: n
        } = t()["features/base/connection"], a = new URL(n.href);
        a.pathname = "/", window.location.assign(a.toString())
    }))
}), e.showThankYou ? 3e3 : 500)
```

**Comment by deleolajide** [comment]
> However, the code is different

The code is different because I have upgraded to latest Jitsi code. See - https://github.com/jitsi/jitsi-meet/blob/a618697e34d947f0cc0d9ee4a0fc79c76fbae5e6/react/features/app/actions.js#L334

**Comment by lochner** [comment]
It then seems that the changes did not reach the minified file that is shipped?

**Comment by deleolajide** [comment]
I am completely lost about what the issue is.

I have made the required changes to the app.bundle.min.js as I usually do and as far as I can tell, confirmed it works for me on both Windows 64 and Linux 64 (Docker). I am not sure of what to do next to fix this issue.  

Have you cleared your browser cache when running the latest nightly build

http://igniterealtime.org/projects/openfire/plugin-archive.jsp?plugin=pade
http://igniterealtime.org/projects/openfire/plugins/1.5.2-SNAPSHOT/pade.jar?snapshot=20211212.202253-3

**Comment by lochner** [comment]
> http://igniterealtime.org/projects/openfire/plugins/1.5.2-SNAPSHOT/pade.jar?snapshot=20211212.202253-3

Really strange situation.&nbsp; The browser cache was cleared for each test of multiple builds since 2021-10-14, the problem was confirmed each time after deployment (including earlier this afternoon for the exact snapshot you refer to above) and the work-around in my original post succeeded each time, confirming the download of the expected file.

Don't worry more about this now.&nbsp; If only I see it, it must be something peculiar about my setup that will become clear eventually, and I'll then report back.&nbsp; If it is a building or packaging problem it will manifest in other ways that might make it easier to pinpoint.

Thank you for your time and for your great effort with Openfire and the Pàdé projects.

**Comment by deleolajide** [comment]
> Thank you for your time and for your great effort with Openfire and the Pàdé projects.

Thanks for the kind words. I will double check again that the changes are in the minified version before I release version 1.5.2 👍

**Comment by gjaekel** [comment]
@lochner Maybe there is another cache in between, i.e. a squid proxy. I also had have to deal with such kind of issues in another context.

**Comment by lochner** [comment]
@gjaekel, that's a gotcha, yes -- I was also concerned about server side caches, but the Nginx/Jetty combination seemed to properly detect updated plug-ins and JS file edits.&nbsp; Thank you for your many contributions to this project!

Thanks @deleolajide, [released version 1.5.2](https://github.com/igniterealtime/openfire-pade-plugin/releases/tag/v1.5.2) was now verified to redirect to the application context.
